lib/experimental/fonts/font-library/class-wp-font-library.php

```diff
@@ -27,6 +27,63 @@ class WP_Font_Library {
 		'woff2' => 'font/woff2',
 	);
 
+	/**
+	 * Font collections.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	public static $collections = array();
+
+	/**
+	 * Register filter to extend the library with font collections.
+	 *
+	 * @since 6.4.0
+	 */
+	public static function register_filters() {
+		add_filter(
+			'register_font_collection',
+			array( 'WP_Font_Library', 'register_font_collection' ),
+			10,
+			2
+		);
+	}
+
+	/**
+	 * Register a new font collection.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $id Font collection id.
+	 * @param array  $config Font collection config options.
+	 * @return WP_Font_Collection|WP_Error A font collection is it was registered succesfully and a WP_Error otherwise.
+	 */
+	public static function register_font_collection( $id, $config ) {
```

Comment:
No need for this to be static.

Reply:
>No need for this to be static.

I respectfully disagree. Let me explain why.

The `WP_Font_Library` is not an object. Rather its public facing methods are a collection of shared functionality.

Rather than separate instances of objects with separate collections, the design is intentional to share one font collection across all of Gutenberg and soon Core.

The design also simplifies usage. How so?

* No instantiation is needed.
* No singleton control is needed.
* Consumers do not need to first get the object instance to then register their font collection.

Thus, IMO the architecture fits Core well and is intentional.

IMO static is appropriate here.

Reply:
I tried to make this avoid `static` and it seems to require more code to do the same. in my opinion, in this case, it also makes the code harder to read  and undersatand.